integration/fixture_test.go

```diff
@@ -88,22 +88,75 @@ func (f *fixture) Curl(url string) (string, error) {
 }
 
 func (f *fixture) CurlUntil(ctx context.Context, url string, expectedContents string) {
+	f.WaitUntil(ctx, fmt.Sprintf("curl(%s)", url), func() (string, error) {
+		return f.Curl(url)
+	}, expectedContents)
+}
+
+func (f *fixture) WaitUntil(ctx context.Context, fnName string, fun func() (string, error), expectedContents string) {
```

Comment:
nit: instead of `fnName`, consider `msg`, because that's more like what it is--and the fatal is something like:
```
Timed out waiting for expected result (<msg>)
expected: <exp>
actual: <act>
```

Reply:
i like that! done